tests/queries_/test_search.py

```diff
@@ -0,0 +1,819 @@
+        cls.create_search_index(
+            Article,
+            "geowithin_location_index",
+            {"mappings": {"dynamic": False, "fields": {"location": {"type": "geo"}}}},
+        )
+
+    def setUp(self):
+        self.article = Article.objects.create(
+            headline="geo", number=2, body="", location={"type": "Point", "coordinates": [40, 5]}
+        )
+        Article.objects.create(
+            headline="geo2", number=3, body="", location={"type": "Point", "coordinates": [-40, -5]}
+        )
+
+    def test_search_geo_within(self):
+        polygon = {
+            "type": "Polygon",
+            "coordinates": [[[30, 0], [50, 0], [50, 10], [30, 10], [30, 0]]],
+        }
+        qs = Article.objects.annotate(
+            score=SearchGeoWithin(
+                path="location",
+                kind="geometry",
+                geo_object=polygon,
+            )
+        )
+        self.delayedAssertCountEqual(qs.all, [self.article])
+
+    def test_constant_score(self):
+        polygon = {
+            "type": "Polygon",
+            "coordinates": [[[30, 0], [50, 0], [50, 10], [30, 10], [30, 0]]],
+        }
+        constant_score = SearchScoreOption({"constant": {"value": 10}})
+        qs = Article.objects.annotate(
+            score=SearchGeoWithin(
+                path="location",
+                kind="geometry",
+                geo_object=polygon,
+                score=constant_score,
+            )
+        )
+        self.delayedAssertCountEqual(qs.all, [self.article])
+        scored = qs.first()
+        self.assertAlmostEqual(scored.score, 10.0, places=2)
+
+
+@skipUnlessDBFeature("supports_atlas_search")
+@unittest.expectedFailure
+class SearchMoreLikeThisTests(SearchUtilsMixin):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.create_search_index(
+            Article,
+            "mlt_index",
+            {
+                "mappings": {
+                    "dynamic": False,
+                    "fields": {"body": {"type": "string"}, "headline": {"type": "string"}},
+                }
+            },
+        )
+        cls.article1 = Article.objects.create(
+            headline="Space exploration", number=1, body="Webb telescope"
+        )
+        cls.article2 = Article.objects.create(
+            headline="The commodities fall",
+            number=2,
+            body="Commodities dropped sharply due to inflation concerns",
+        )
+        Article.objects.create(
+            headline="irrelevant",
+            number=3,
+            body="This is a completely unrelated article about cooking",
+        )
+
+    def test_search_more_like_this(self):
+        like_docs = [
+            {"headline": self.article1.headline, "body": self.article1.body},
+            {"headline": self.article2.headline, "body": self.article2.body},
+        ]
+        like_docs = [{"body": "NASA launches new satellite to explore the galaxy"}]
```

Comment:
I think this gets overridden. 

Also, should `article2` pop up as a valid result?
```suggestion
        like_docs = [{"body": "NASA launches new satellite to explore the galaxy"}]
```